client/src/client.rs

```diff
@@ -313,8 +400,41 @@ impl Client {
             )
     }
 
+    /// Lower-level Query API entry point. Prepares an http-request and returns it with an http-response handler.
+    ///
+    /// # Errors
+    /// Fails if query signing or request building fails.
+    fn prepare_query_request<R, B>(
+        &self,
+        request: R,
+        pagination: Pagination,
+    ) -> Result<(B, QueryResponseHandler<R>)>
+    where
+        R: Query + Into<QueryBox> + Debug,
+        <R::Output as TryFrom<Value>>::Error: Into<eyre::Error>,
+        B: RequestBuilder,
+    {
+        let pagination: Vec<_> = pagination.into();
+        let request = QueryRequest::new(request.into(), self.account_id.clone());
+        let request: VersionedSignedQueryRequest = self.sign_query(request)?.into();
+
+        Ok((
+            B::build(
+                HttpMethod::POST,
+                format!("{}/{}", &self.torii_url, uri::QUERY),
+                request.encode_versioned(),
+                pagination,
+                self.headers.clone(),
+            )?,
+            QueryResponseHandler(PhantomData),
```

Comment:
Looks strange... Maybe add `new()` or `default()` to `QueryResponseHandler`?

Reply:
Agree